util/conditions/v1beta2/setter_test.go

```diff
@@ -23,95 +23,76 @@ import (
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/internal/test/builder"
 )
 
-func TestSetAll(t *testing.T) {
+func TestSet(t *testing.T) {
 	now := metav1.Now().Rfc3339Copy()
 
-	conditions := []metav1.Condition{
-		{
-			Type:               "fooCondition",
-			Status:             metav1.ConditionTrue,
-			ObservedGeneration: 10,
-			LastTransitionTime: now,
-			Reason:             "FooReason",
-			Message:            "FooMessage",
-		},
+	condition := metav1.Condition{
+		Type:               "fooCondition",
+		Status:             metav1.ConditionTrue,
+		ObservedGeneration: 0, // NOTE: the is a dedicated tests about inferring ObservedGeneration.
+		LastTransitionTime: now,
+		Reason:             "FooReason",
+		Message:            "FooMessage",
 	}
 
-	cloneConditions := func() []metav1.Condition {
-		ret := make([]metav1.Condition, len(conditions))
-		copy(ret, conditions)
-		return ret
+	cloneCondition := func() metav1.Condition {
```

Comment:
nit: if we make the change so that Set takes a *metav1.Condition we can simply use DeepCopy here and get rid of cloneCondition

Reply:
discussed above